src/platform_util.cpp

```diff
@@ -74,8 +74,8 @@ namespace libtorrent {
 			if (rl.rlim_cur == rlim_infinity)
 				return std::numeric_limits<int>::max();
 
-			return rl.rlim_cur <= std::numeric_limits<int>::max()
-				? int(rl.rlim_cur) : std::numeric_limits<int>::max();
+			return static_cast<int>(rl.rlim_cur) <= std::numeric_limits<int>::max()
```

Comment:
the reason this comparison is here is because `rl.rlim_cur` may be greater than the largest value that can be represented by `int`. If it is, this function just returns that largest possible value. Since this function returns an `int`, it cannot return the true (larger) number in that case.

By casting the `rl.rlim_cur`, you effectively truncate that value *first*, making the test meaningless. to make the test meaningful, you'd have to cast the *right* hand side to `rlim_t`.